### PR TITLE
Update the OpenTelemetry SDK version to 1.27.0

### PR DESCRIPTION
**Pull request by opentelemetrybot**
Update the OpenTelemetry SDK version to `1.27.0`.

### CODE REVIEW

**Comment by trask** [comment]
(I'm working on fixing)

**Comment by trask** [comment]
(reminder for myself, need to deal with opentelemetry-api-logs instrumentation)

File: conventions/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-testing.gradle.kts [hunk 1]
```diff
@@ -83,7 +79,7 @@ class JavaagentTestArgumentsProvider(
     "-Dotel.javaagent.testing.javaagent-jar-path=${agentShadowJar.absolutePath}",
     "-Dotel.javaagent.experimental.initializer.jar=${shadowJar.absolutePath}",
     "-Dotel.javaagent.testing.additional-library-ignores.enabled=false",
-    "-Dotel.javaagent.testing.fail-on-context-leak=${findProperty("failOnContextLeak") != false}",
+    "-Dotel.javaagent.testing.fail-on-context-leak=${findProperty("failOnContextLeak")!=false}",
```
Comment (trask) [inline conventions/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-testing.gradle.kts:82]:
```suggestion
    "-Dotel.javaagent.testing.fail-on-context-leak=${findProperty("failOnContextLeak") != false}",
```

**Review by trask — APPROVED** [review]

File: instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java [hunk 2]
```diff
@@ -83,16 +82,14 @@ public SdkLoggerProvider sdkLoggerProvider(
...
           .forEach(
               loggerExporter ->
                   loggerProviderBuilder.addLogRecordProcessor(
                       BatchLogRecordProcessor.builder(loggerExporter).build()));
 
-      SdkLoggerProvider loggerProvider = loggerProviderBuilder.build();
-      GlobalLoggerProvider.set(loggerProvider);
-
-      return loggerProvider;
+      return loggerProviderBuilder.build();
```
Comment (mateuszrzeszutek) [inline instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java:92]: The Spring starter does not set `GlobalOpenTelemetry`, we'll have to do that if we want logs to work (and add `resetForTests()` calls to the test classes)

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/messaging/MessagingAttributesGetter.java [hunk 3]
```diff
@@ -22,9 +22,6 @@
   @Nullable
   String getSystem(REQUEST request);
 
-  @Nullable
-  String getDestinationKind(REQUEST request);
```
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/messaging/MessagingAttributesGetter.java:26]: WDYT about adding a `default ... { return null; }` implementation for a deprecation cycle?

File: instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java (hunk 2 above)
Comment (trask) [reply instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java:92]: thx, I totally missed this 🙈

File: instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java [hunk 4]
```diff
@@ -143,6 +141,9 @@ public OpenTelemetry openTelemetry(
 
       ContextPropagators propagators = propagatorsProvider.getIfAvailable(ContextPropagators::noop);
 
+      // global is needed for logging appenders
+      GlobalOpenTelemetry.set(OpenTelemetrySdk.builder().setLoggerProvider(loggerProvider).build());
```
Comment (trask) [inline instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java:145]: this is my proposed compromise related to https://github.com/open-telemetry/opentelemetry-java/blob/main/docs/sdk-configuration.md#the-global-instance-of-the-sdk

Comment (mateuszrzeszutek) [reply instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java:145]: Hmm, this looks a bit hacky -- at the same time, I don't think I have any strong opinion on how we should solve this problem. Let's merge it as it is, and discuss the global issue on the SIG meeting

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by laurit — APPROVED** [review]

**Pull request merged**